client/my-sites/site-settings/settings-p2/preapproved-domains.jsx

```diff
@@ -0,0 +1,264 @@
+	}
+
+	const getFormField = () => {
+		return fields?.p2_preapproved_domains || [];
+	};
+
+	const setFormField = ( domains ) => {
+		updateFields( { [ SETTING_KEY_PREAPPROVED_DOMAINS ]: domains } );
+	};
+
+	const handleSubmitButtonClick = ( event ) => {
+		if ( ! isValidating ) {
+			handleSubmitForm( event );
+		}
+	};
+
+	const handleDomainsToggle = () => {
+		// Clear field if toggling off. If toggling on, it should have
+		// been empty to start with.
+		setFormField( [] );
+
+		setIsToggledOn( ! isToggledOn );
+	};
+
+	const refreshValidation = ( valid = [], invalid = {} ) => {
+		setInvalidTokens( invalid );
+		setValidTokens( valid );
+
+		const invalidTokenKeys = Object.keys( invalidTokens );
+		setError( error || ( invalidTokenKeys.length > 0 && invalidTokenKeys[ 0 ] ) );
+	};
+
+	const validateTokens = async ( tokens ) => {
+		setIsValidating( true );
+
+		if ( tokens.length < 1 ) {
+			refreshValidation( [], {} );
+			return;
+		}
+
+		// Prior to sending off a validation API request,
+		// first quickly remove old, now-deleted tokens from validation lists.
+		const updatedInvalidTokens = pickBy( invalidTokens, ( _, key ) => {
+			return tokens.includes( key );
+		} );
+		setInvalidTokens( updatedInvalidTokens );
+		setError( tokens.includes( error ) && error );
+
+		const updatedValidTokens = filter( validTokens, ( token ) => {
+			return tokens.includes( token );
+		} );
+		setValidTokens( updatedValidTokens );
+
+		try {
+			const { success, errors } = await wpcom.req.get(
+				{
+					path: `/p2/hub-settings/domains/validate`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{ domains: tokens }
+			);
+
+			refreshValidation( success, errors );
+
+			recordTracksEvent( 'calypso_p2_settings_validate_preapproved_domains_valid' );
+		} catch ( e ) {
+			recordTracksEvent( 'calypso_p2_settings_validate_preapproved_domains_invalid' );
+		} finally {
+			setIsValidating( false );
+		}
+	};
+
+	const normalizeTokens = ( rawTokens ) => {
+		const tokens = rawTokens.map( ( value ) => {
+			if ( 'object' === typeof value ) {
+				return value.value;
+			}
+			return value;
+		} );
+
+		return tokens;
+	};
+
+	const onTokensChange = ( rawTokens ) => {
+		const tokens = normalizeTokens( rawTokens );
+		setFormField( tokens );
+		validateTokens( tokens );
+	};
+
+	const getTokensWithStatus = () => {
+		const rawTokens = getFormField();
+
+		const tokens = rawTokens.map( ( token ) => {
+			if ( invalidTokens && invalidTokens[ token ] ) {
+				return {
+					status: 'error',
+					value: token,
+					tooltip: invalidTokens[ token ],
+					onMouseEnter: () => setError( token ),
+				};
+			}
+
+			if ( ! includes( validTokens, token ) ) {
+				return {
+					value: token,
+					status: 'validating',
+				};
+			}
+
+			return token;
+		} );
+
+		return tokens;
+	};
+
+	const classes = classNames( 'site-settings__p2-preapproved-domains', {
+		'is-loading': isRequestingSettings,
+	} );
+
+	return (
+		<div className={ classNames( classes ) }>
```

Comment:
I think we can just include the:

```
<div className={ classNames( 'site-settings__p2-preapproved-domains', {
		'is-loading': isRequestingSettings,
	} ) } >
```
here without having to make an intermediate variable.

Reply:
Done!